access/types/forms.py

```diff
@@ -67,6 +59,14 @@ def __init__(self, *args, **kwargs):
 
             # Randomly pick fields to include.
             if "pick_randomly" in group:
+                # Check that sample is unmodified in a randomized form.
+                if not args[0] is None:
+                    nonce = args[0].get('_nonce', '')
+                    sample = args[0].get('_sample', '')
+                    if self.samples_hash(nonce, sample) != args[0].get('_checksum', ''):
+                        raise PermissionDenied('Invalid checksum')
+                    post_samples = sample.split('/')
+
                 self.randomized = True
                 if not args[0] is None:
```

Comment:
There are two if statements with the same condiftion `if not args[0] is None`. I also prefer `is not` over `not X is ...`